### ISSUE TITLE
Duplicate middleware name 'NewRelicHeader' but not using agent directly

### ISSUE THREAD

**Issue by decompil3d** [newrelic/serverless-newrelic-lambda-layers#347]
Hi folks,

I'm using the latest version of this plugin (3.4.3) and see the latest version of the NodeJS lambda layer loading on my Lambda (NewRelicNodeJS18XARM64 version 23). But I am seeing the same problem as #293:

```
    "err": {
        "type": "Error",
        "message": "Duplicate middleware name 'NewRelicHeader'",
        "stack": "Error: Duplicate middleware name 'NewRelicHeader'
            at Object.add (/var/task/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:130:31)
            at Object.applyToStack (/opt/nodejs/node_modules/@newrelic/aws-sdk/lib/v3/smithy-client.js:45:19)
            at Object.use (/var/task/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:167:20)
            at SecretsManagerClient.wrapShim (/opt/nodejs/node_modules/@newrelic/aws-sdk/lib/v3/util.js:33:26)
            at new WrappedClass (/opt/nodejs/node_modules/newrelic/lib/shim/shim.js:2064:30)
            at new SecretsManagerClient (/var/task/node_modules/@aws-sdk/client-secrets-manager/dist-cjs/SecretsManagerClient.js:26:9)
            at getSecret (/var/task/v1/index.js:794:18)
            at RedactedClass.redactedMethod1 (/var/task/v1/index.js:813:23)
            at RedactedClass.redactedMethod2 (/var/task/v1/index.js:834:28)
            at RedactedClass.redactedMethod3 (/var/task/v1/index.js:737:31)"
    }
```

Could it possibly be caused by https://github.com/newrelic/node-newrelic/issues/1565? I'm not sure how to verify which version of the agent is within the layer.

I do not have the `newrelic` package installed at all in my app. Just using the plain vanilla Serverless plugin integration. But this error is breaking my application repeatedly.

**Comment by hatzi** [comment]
I had the same issue, then realised I had included `@aws-sdk/client-secrets-manager` as a dependency, which was then being bundled as part of the package. Moving this package under `devDependencies`, and therefore no longer bundled as part of my package did the trick for me

**Comment by mikeshot** [comment]
We are having the same issue when using `serverless-newrelic-lambda-layers`

**Environment**

- AWS Lamba
- NodeJS v18.x
- AWS Lambda layer `arn:aws:lambda:eu-west-1:451483290750:layer:NewRelicNodeJS18X:29` (the error starts at version `NewRelicNodeJS18X:28`)
- Relevant dependencies
    - "@aws-sdk/client-dynamodb": "^3.358.0",
    - "@aws-sdk/client-lambda": "^3.358.0",
    - "@aws-sdk/client-rds-data": "^3.358.0",
    - "@aws-sdk/client-ssm": "^3.358.0",
    - "@aws-sdk/lib-dynamodb": "^3.358.0",

**Error**
``` 
{
  "timestamp": "2023-06-26T15:10:37.505Z",
  "level": "error",
  "message": "[RdsDataService] ERROR: Duplicate middleware name 'NewRelicHeader'",
  "data": "Error: Duplicate middleware name 'NewRelicHeader'
      at Object.add (/var/runtime/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:130:31)
      at Object.applyToStack (/opt/nodejs/node_modules/@newrelic/aws-sdk/lib/v3/smithy-client.js:45:19)
      at Object.use (/var/runtime/node_modules/@aws-sdk/middleware-stack/dist-cjs/MiddlewareStack.js:167:20)
      at RDSData.wrapShim (/opt/nodejs/node_modules/@newrelic/aws-sdk/lib/v3/util.js:33:26)
      at new WrappedClass (/opt/nodejs/node_modules/newrelic/lib/shim/shim.js:2085:30)
      at new RDSDataClient (/var/runtime/node_modules/@aws-sdk/client-rds-data/dist-cjs/RDSDataClient.js:23:9)
      at new RDSData (/var/runtime/node_modules/@aws-sdk/client-rds-data/dist-cjs/RDSData.js:11:1)"
}
```

**Workaround**
As a workaround, we used `arn:aws:lambda:eu-west-1:451483290750:layer:NewRelicNodeJS18X:27` and is working fine.

**Comment by nthornton2010** [comment]
This appears to be fixed in version 32. We ran into this on version 30 and forced ourselves to version 27. On 32 its no longer happening.

**Comment by bizob2828** [comment]
Duplicate of https://github.com/newrelic/node-newrelic-aws-sdk/issues/211 which we have fixed and released in `@newrelic/aws-sdk` `v5.0.3`
